### PR TITLE
[SDA-8351] Add resources for clusters/:id/control_plane/upgrade_policies/:id

### PR DESCRIPTION
**Pull request by gdbranco**
Related issue: https://issues.redhat.com/browse/SDA-8351

### CODE REVIEW

File: model/clusters_mgmt/v1/control_plane_upgrade_policy_type.model [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	// Cluster ID this upgrade policy for control plane is defined for.
+	ClusterID String
+
+	// Schedule cron expression that defines automatic upgrade scheduling.
+	Schedule String
+
+	// Schedule type can be either "manual" (single execution) or "automatic" (re-occurring).
+	ScheduleType String
+
+	// Upgrade type specify the type of the upgrade. Can be "OSD" or "CVE".
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/control_plane_upgrade_policy_type.model:28]: Can be only "ControlPlane" at this time.

File: model/clusters_mgmt/v1/control_plane_upgrade_policy_type.model [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	// Version is the desired upgrade version.
+	Version String
+
+	// Next time the upgrade should run.
+	NextRun Date
+
+	// Indicates if minor version upgrades are allowed for automatic upgrades (for manual it's always allowed).
+	EnableMinorVersionUpgrades Boolean
+
+    // Timestamp for creation of resource
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/control_plane_upgrade_policy_type.model:40]: NP: alignment is off

**Review by tzvatot — CHANGES_REQUESTED** [review]

File: model/clusters_mgmt/v1/control_plane_upgrade_policy_type.model (hunk 1 above)
Comment (gdbranco) [reply model/clusters_mgmt/v1/control_plane_upgrade_policy_type.model:28]: hmm, considering this endpoint is specific for this. Does it make sense to have this attribute at all?

Comment (tzvatot) [reply model/clusters_mgmt/v1/control_plane_upgrade_policy_type.model]: The purpose of this is to later support other types, like CVE upgrades. I suggest to keep it at this time, to allow this kind of flexibility.

**Review by tzvatot — APPROVED** [review]

**Pull request merged**